### PR TITLE
DTSPO-172 Create module to reference common tags  

### PR DESCRIPTION
**Pull request by mokainos**
https://tools.hmcts.net/jira/browse/DTSPO-172

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  type        = string
+}
+variable "businessArea" {
+  description = "Enter name of the business area this application belong to."
+  type        = string
+}
+variable "builtFrom" {
+  description = "Enter name of the GitHub repository this application is being build from."
+  type        = string
+}
```
Comment (timja) [inline variables.tf:16]:
```suggestion
}

```

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ptl      = "High"
+    preview  = "Low"
+    ldata    = "High"
+    sandbox  = "Low"
+  }
+
+  env_mapping = {
+    production  = ["ptl", "prod"]
+    development = ["dev", "preview"]
+    staging     = ["ldata", "stg", "aat"]
```
Comment (timja) [inline main.tf:38]:
```suggestion
    staging     = ["ldata", "stg", "aat", "nle"]
```

**Review by timja — CHANGES_REQUESTED** [review]
README needs re-generating, output is wrong and so is description

**Comment by mokainos** [comment]
> README needs re-generating, output is wrong and so is description

updated the file

File: README.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
-# terraform-module-common-tags
-Common Tags
+
```
Comment (timja) [inline README.md:1]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,30 @@
...
+# Terraform module to reference common tags.
```
Comment (timja) [inline README.md:2]: a heading is not a sentence
```suggestion
# Terraform module to reference common tags
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,30 @@
...
+
+## Tagging policy documentation.
```
Comment (timja) [inline README.md:4]:
```suggestion
## Tagging policy documentation
```

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * https://tools.hmcts.net/confluence/display/DCO/Tagging+v0.4
+ *
+ *
+ */
+
+locals {
+  common_tags = {
+    application  = var.application
+    businessArea = var.businessArea
```
Comment (timja) [inline main.tf:15]: can we validate this is one of the allowed values?

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  validation {
+    condition = can(regex("^probate|^divorce|^core", var.application))
+    error_message = "Name of the application is one of 'probate' / 'divorce' / 'core'."
+  }
+}
+variable "businessArea" {
+  description = "Name of the business area this application belongs to, one of CFT / Crime / Cross-Cutting."
+  type        = string
+  validation {
+    condition = can(regex("^CFT|^Crime|^Cross-Cutting", var.businessArea))
```
Comment (timja) [inline variables.tf:17]: as discussed this should be moved to configuration (yml file) rather than an input

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,24 @@
+variable "environment" {
+  description = "Name of the environment to deploy the resource"
+  type        = string
+}
+variable "application" {
+  description = "Name of the application, e.g. 'probate', 'divorce', use 'core' for platform infrastructure"
+  type        = string
+  validation {
+    condition = can(regex("^probate|^divorce|^core", var.application))
```
Comment (timja) [inline variables.tf:9]: as discussed this should be moved to configuration (yml file) rather than an input

**Review by timja — CHANGES_REQUESTED** [review]

File: team-config.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+sds-platform:
+  team: "platform operations"
+  namespace: ""
+  slack:
+    contact_channel: "#platops-help"
+    build_notices_channel: "#platops-build-notices"
+  tags:
+    application: "core"
```
Comment (timja) [inline team-config.yml:8]: application and business area should be explained somehow, maybe a comment at the top of the file saying businessArea must be one of ...

and that application is normally the team name except for 'platform' which is core

**Review by timja — APPROVED** [review]

**Pull request merged**